### PR TITLE
Resend unacked fix

### PR DESCRIPTION
**Pull request by kanes115**

### CODE REVIEW

**Comment by kanes115** [comment]
For now - it looks strange - we get a list of stanzas that ARE messages and ARE in a proper order but somehow different messages get assertet to each other
Failed test:
http://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/1601/3809/mysql_redis.19.3/big/ct_run.test@travis-job-63b72be2-55f1-4835-9323-e3a67653e3a3.2017-12-12_16.42.46/ejabberd_tests.tests.sm_SUITE.logs/run.2017-12-12_16.54.29/sm_suite.resend_unacked_on_reconnection.7123.html

Passed test:
http://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/1601/3809/odbc_pgsql_mnesia.19.3/big/ct_run.test@travis-job-4f3ed6bf-311f-4173-b788-3185aee46cdb.2017-12-12_16.45.10/ejabberd_tests.tests.sm_SUITE.logs/run.2017-12-12_17.01.18/sm_suite.resend_unacked_on_reconnection.7198.html

**Comment by kanes115** [comment]
Situation where the 3rd message was a presence (Alice to Alice available) instead of message 3 (mysql-redis). NOTICE that the <<"msg-3">> message would come just after this presence:

[ct-log](http://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/1601/3824/mysql_redis.19.3/big/ct_run.test@travis-job-8ac77697-fac4-410c-a7e6-5333aba3517d.2017-12-14_12.43.49/ejabberd_tests.tests.sm_SUITE.logs/run.2017-12-14_12.55.30/sm_suite.resend_unacked_on_reconnection.7318.html)

Same situation (odbc_pgsql_mnesia DB=pgsql):

[ct-log](http://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/1601/3824/odbc_pgsql_mnesia.19.3/big/ct_run.test@travis-job-25bc394b-6eb7-4f90-9390-c5033bfbcb80.2017-12-14_12.50.35/ejabberd_tests.tests.sm_SUITE.logs/run.2017-12-14_13.07.39/sm_suite.resend_unacked_on_reconnection.6835.html)

Exact same situation (riak_mnesia DB=riak):

[ct-log](http://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/1601/3824/riak_mnesia.18.3/big/ct_run.test@travis-job-562b448b-e94a-4955-ad54-1095159cdbd5.2017-12-14_13.17.56/ejabberd_tests.tests.sm_SUITE.logs/run.2017-12-14_13.33.41/sm_suite.resend_unacked_on_reconnection.2379.html)

**Comment by kanes115** [comment]
Conclusion1: It does not depend on the preset/database

**Comment by kanes115** [comment]
Conclusion2: We receive unexpected presence before getting the message (3rd one but maybe it's possible to get it before another one)

**Comment by kanes115** [comment]
Now all sm_SUITE has always passed. I am, though, not sure why other suites failed.

**Comment by kanes115** [comment]
Hypothesis 1 - not true - client should send this element but it is not the reason of the test failure.
The test still fails after this fix

**Comment by kanes115** [comment]
Observation3: In [this](https://travis-ci.org/esl/MongooseIM/jobs/318030710) build we got this log:
```
2017-12-18 12:37:29.628 [warning] <0.3809.1>@ejabberd_c2s:do_resume_session:3125 resumption error (invalid response from undefined)
2017-12-18 12:37:29.634 [warning] <0.3816.1>@ejabberd_c2s:do_resume_session:3132 no previous session with stream id <<"wrong-sid">>
```

**Comment by michalwski** [comment]
This seams OK, I remember there is a test cases checking if you get error when you try to resume a session which doesn't exists.

File: test.disabled/ejabberd_tests/tests/sm_SUITE.erl [hunk 1]
```diff
@@ -880,7 +902,14 @@ given_fresh_spec(Config, User) ->
 
 given_fresh_user(Config, UserName) ->
     Spec = given_fresh_spec(Config, UserName),
-    given_fresh_user_with_spec(Spec).
+    case proplists:is_defined(connection_steps, Config) of
+        true ->
+            Spec2 = [{connection_steps, proplists:get_value(connection_steps, Config)} | Spec];
```
Comment (arcusfelis) [inline test.disabled/ejabberd_tests/tests/sm_SUITE.erl]: Should be 

```
Spec2 = case ... end
```

assigning inside case is some bad practice (most likely, banned in our coding standards).

**Comment by kanes115** [comment]
This was never meant to be merged. Not needed anymore.

**Pull request closed without merging**